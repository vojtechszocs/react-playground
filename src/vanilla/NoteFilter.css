.root {
  display: inline-block;
  float: right;
  border: 2px dashed grey;
  padding: 1em;
  margin-left: 1em;
}

.input {
  margin-left: 0.5em;
  margin-right: 0.5em;
  width: 20em;
}
