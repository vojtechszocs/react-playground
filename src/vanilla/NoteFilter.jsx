import React from 'react'
import PropTypes from 'prop-types'

import style from './NoteFilter.css'

const NoteFilter = ({ value, onChange, onReset }) => (
  <div className={style.root}>

    <span>Find notes:</span>

    <input type='text' className={style.input}
      placeholder='Filter by text or tags'
      value={value}
      onChange={event => { onChange(event.target.value) }} />

    <button onClick={onReset} disabled={!value}>
      Reset
    </button>

  </div>
)

NoteFilter.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired
}

export default NoteFilter
