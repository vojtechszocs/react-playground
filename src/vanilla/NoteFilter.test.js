/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import NoteFilter from './NoteFilter'
import style from './NoteFilter.css'

describe('NoteFilter', () => {

  test('renders correctly', () => {
    expect(renderer.create(
      <NoteFilter value='test'
        onChange={() => {}}
        onReset={() => {}} />
    )).toMatchSnapshot()

    expect(renderer.create(
      <NoteFilter value=''
        onChange={() => {}}
        onReset={() => {}} />
    )).toMatchSnapshot()
  })

  test('reset button interaction', () => {
    const onResetMock = jest.fn()

    const wrapper = shallow(
      <NoteFilter value='test'
        onChange={() => {}}
        onReset={onResetMock} />
    )

    wrapper.find('button').first().simulate('click')
    expect(onResetMock).toHaveBeenCalled()
  })

  test('input field interaction', () => {
    const onChangeMock = jest.fn()

    const wrapper = shallow(
      <NoteFilter value='test'
        onChange={onChangeMock}
        onReset={() => {}} />
    )

    wrapper.find(`input.${style.input}`).simulate('change', {
      target: {
        value: 'foo'
      }
    })
    expect(onChangeMock).toHaveBeenCalledWith('foo')
  })

})
