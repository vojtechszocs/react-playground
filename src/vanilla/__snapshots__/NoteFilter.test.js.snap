// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteFilter renders correctly 1`] = `
<div
  className="root"
>
  <span>
    Find notes:
  </span>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Filter by text or tags"
    type="text"
    value="test"
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;

exports[`NoteFilter renders correctly 2`] = `
<div
  className="root"
>
  <span>
    Find notes:
  </span>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Filter by text or tags"
    type="text"
    value=""
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
